openedx/core/djangoapps/user_api/preferences/api.py

```diff
@@ -110,7 +112,8 @@ def update_user_preferences(requesting_user, update, username=None):
         PreferenceUpdateError: the operation failed when performing the update.
         UserAPIInternalError: the operation failed due to an unexpected error.
     """
-    existing_user = _get_user(requesting_user, username)
+    if not existing_user:
```

Comment:
@wajeeha-khalid Your approach sounds good, but do make sure to use basestring for the insinstance call, as @jcdyer suggested.

How about renaming _get_user to _get_authorized_user to make it clear that it also does authorization? Or if you like @jcdyer's suggestion, that's fine as well (though you will need to add the additional call everyplace that _get_user is called).